ngraph/test/type_prop/logical_not.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+namespace {
+    void type_check(const ngraph::element::Type& type) {
+        auto input = make_shared<op::Parameter>(type, Shape{1, 3, 6});
+        auto logical_not = make_shared<op::v1::LogicalNot>(input);
+
+        ASSERT_EQ(logical_not->get_element_type(), type);
+    }
+}
+
+TEST(type_prop, logical_not_i32)
```

Comment:
As mentioned in the spec the type of input should be `boolean`. There are few comments regarding precision, to restrict type to only `boolean`:

https://github.com/openvinotoolkit/openvino/blob/aa3645cb53564d2a58856922a80cc34182f91d93/ngraph/core/src/op/not.cpp#L31-L40

https://github.com/openvinotoolkit/openvino/blob/3826a0d08df35c8c936a1828420b8843ffd14cb0/ngraph/test/type_prop/binary_elementwise.cpp#L544-L558

So it sounds reasonable to add a `NODE_VALIDATION_CHECK` for `boolean` type in the input in `validate_and_infer_types` and remove other precisions in [`evaluate`](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/not.cpp#L60) and [`has_evaluate`](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/op/not.cpp#L87) methods